csharp/extractor/Semmle.Extraction/Context.cs

```diff
@@ -135,10 +159,13 @@ private void CheckEntityHasUniqueLabel(string id, ICachedEntity entity)
 
         public Label GetNewLabel() => new Label(GetNewId());
 
-        private Entity CreateNonNullEntity<Type, Entity>(ICachedEntityFactory<Type, Entity> factory, Type init) where Entity : ICachedEntity
+        public Entity CreateNonNullEntity<Type, Entity>(ICachedEntityFactory<Type, Entity> factory, Type init)
+            where Entity : ICachedEntity
         {
+            if (init is null) throw new ArgumentException("Unexpected null value", nameof(init));
+
             if (objectEntityCache.TryGetValue(init, out var cached))
-                return (Entity)cached;
+                return (Entity)cached!;
```

Comment:
Why is the `!` needed? I would have expected `cached` to be non-`null` given the type `Dictionary<object, ICachedEntity>` of `objectEntityCache`.